### PR TITLE
added actionbar and overflow menu with settings

### PR DESCRIPTION
**Pull request by shubhangb97**

### CODE REVIEW

File: app/src/main/res/menu/settings_menu.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (unstablebrainiac) [inline app/src/main/res/menu/settings_menu.xml]: Rename this file to `main.xml`

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -1,4 +1,5 @@
 <resources>
     <string name="app_name">Tum Tum Tracker</string>
     <string name="title_activity_maps">Map</string>
+    <string name="settings">settings</string>
```
Comment (unstablebrainiac) [inline app/src/main/res/values/strings.xml]: Settings with a capital 'S'

File: app/src/main/java/in/ac/iitb/gymkhana/tumtumtracker/MapsActivity.java [hunk 3]
```diff
@@ -22,6 +26,9 @@ protected void onCreate(Bundle savedInstanceState) {
         SupportMapFragment mapFragment = (SupportMapFragment) getSupportFragmentManager()
                 .findFragmentById(R.id.map);
         mapFragment.getMapAsync(this);
+        getSupportActionBar().setTitle("Tum Tum Tracker");
```
Comment (unstablebrainiac) [inline app/src/main/java/in/ac/iitb/gymkhana/tumtumtracker/MapsActivity.java]: Fetch app name from `strings.xml` instead of hard coding

**Review by unstablebrainiac — CHANGES_REQUESTED** [review]
Just 3 small changes, and then we'll merge

**Comment by shubhangb97** [comment]
Done the changes

**Comment by shubhangb97** [comment]
added button to zoom to current location

File: app/src/main/java/in/ac/iitb/gymkhana/tumtumtracker/MapsActivity.java [hunk 4]
```diff
@@ -37,10 +49,56 @@ protected void onCreate(Bundle savedInstanceState) {
...
+        }
         // Add a marker in Sydney and move the camera
-        LatLng sydney = new LatLng(-34, 151);
-        mMap.addMarker(new MarkerOptions().position(sydney).title("Marker in Sydney"));
-        mMap.moveCamera(CameraUpdateFactory.newLatLng(sydney));
+        // You need to change these values for current location
+        // LatLng sydney = new LatLng(-34, 151);
+        //mMap.addMarker(new MarkerOptions().position(sydney).title("Marker in Sydney"));
+        //mMap.moveCamera(CameraUpdateFactory.newLatLng(sydney));
+        if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
```
Comment (unstablebrainiac) [inline app/src/main/java/in/ac/iitb/gymkhana/tumtumtracker/MapsActivity.java]: This does not work as expected on my phone. I get a permission prompt and on clicking _Allow_, there are 2 consecutive toasts _Need Permission_ and _Restart after giving permission_ in that order. Ideally, if the user selects _Allow_, he shouldn't have to restart the app to use the functionality.

**Review by unstablebrainiac — CHANGES_REQUESTED** [review]

File: app/src/main/java/in/ac/iitb/gymkhana/tumtumtracker/MapsActivity.java (hunk 4 above)
Comment (shubhangb97) [reply app/src/main/java/in/ac/iitb/gymkhana/tumtumtracker/MapsActivity.java]: After giving the permission if we restart the app it works pefectly. I will try to correct it,looks like an issue in the API

**Comment by shubhangb97** [comment]
resolved the permission restart issue

File: app/src/main/java/in/ac/iitb/gymkhana/tumtumtracker/MapsActivity.java [hunk 5]
```diff
@@ -10,7 +20,9 @@
 import com.google.android.gms.maps.model.LatLng;
 import com.google.android.gms.maps.model.MarkerOptions;
 
-public class MapsActivity extends FragmentActivity implements OnMapReadyCallback {
+import static android.support.v4.content.PermissionChecker.PERMISSION_GRANTED;
```
Comment (unstablebrainiac) [inline app/src/main/java/in/ac/iitb/gymkhana/tumtumtracker/MapsActivity.java]: What is this for?

**Review by unstablebrainiac — APPROVED** [review]

File: app/src/main/java/in/ac/iitb/gymkhana/tumtumtracker/MapsActivity.java (hunk 5 above)
Comment (shubhangb97) [reply app/src/main/java/in/ac/iitb/gymkhana/tumtumtracker/MapsActivity.java]: Thats a static int constant  to check if permission was granted or not used in onRequestPermissionsResult() method

Comment (unstablebrainiac) [reply app/src/main/java/in/ac/iitb/gymkhana/tumtumtracker/MapsActivity.java]: You can use it without the import. Try doing that

File: app/src/main/java/in/ac/iitb/gymkhana/tumtumtracker/MapsActivity.java [hunk 6]
```diff
@@ -37,10 +50,49 @@ protected void onCreate(Bundle savedInstanceState) {
...
+            mMap.setMyLocationEnabled(true);
+            mMap.getUiSettings().setMyLocationButtonEnabled(true);
+            mMap.getUiSettings().setZoomGesturesEnabled(true);
+        }
+    }
+    public void onRequestPermissionsResult (int requestCode,
+                                           String[] permissions,
+                                           int[] grantResults)
+    {
+        if(grantResults[0]==getPackageManager().PERMISSION_GRANTED)
```
Comment (unstablebrainiac) [inline app/src/main/java/in/ac/iitb/gymkhana/tumtumtracker/MapsActivity.java]: `static` variable hai, `PackageManager.PERMISSION_GRANTED` will do

Comment (shubhangb97) [reply app/src/main/java/in/ac/iitb/gymkhana/tumtumtracker/MapsActivity.java]: ab ho gaya Sorry androidStudio had auto imported those constants previously

**Pull request merged**